Random continuous integration build failures

### IINA Version

1.4.1

### Release Channel

Other

### macOS Version

26.4

### Steps to Reproduce

Run the continous integration build for a recently created pull request.

### Expected Behavior

The CI build passes.

### Actual Behavior

 Randomly the CI build will fail. Thanks to the excellent error reporting recently added to `download_libs.sh` by @DavidAGInnovation we now know why it is failing:

```text
Downloading yt-dlp...
yt-dlp downloaded
Downloading latest plugin release for iina/plugin-online-media...
GitHub API returned HTTP 403 for iina/plugin-online-media.
API rate limit exceeded for 13.105.117.183. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
Failed to fetch the latest plugin asset for iina/plugin-online-media.
Error: Process completed with exit code 1.
```

Recently `download_libs.sh` was enhanced to download the official plugins. That use of the GitHub API _sometimes_ triggers this GitHub rate limit. Maybe it depends upon what other jobs are running on the same GitHub CI build server?

So far all the failures I investigated happen in the same spot, right after the script downloads `yt-dlp` and starts the download the first plugin.

Not sure how to fix this. One way would be to add an option to skip downloading plugins and forgo that during CI builds. Need to see what others think…

### Crash Report or Process Sample

_No response_

### Logs

_No response_

### Sample Files

_No response_

### How often can you reproduce the issue

Fairly often.

### Required Actions


- [x] I tried my best to search from the existing issues (including closed issues) and didn't find duplicated issues.

### Optional Actions


- [ ] I tested with the Nightly build and confirmed the issue is not fixed.
- [ ] I tested with the latest version of mpv and confirmed mpv doesn't have this issue.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -18,7 +18,9 @@ jobs:
       - uses: actions/checkout@v6
 
       - name: Install dependencies
-        run: ./other/download_libs.sh
+        run: ./other/download_libs.sh $DOWNLOAD_LIBS_ARGS
+        env:
+          DOWNLOAD_LIBS_ARGS: ${{ github.event_name == 'pull_request' && '--skip-plugins' || '' }}
 
       - name: Build
         run: xcodebuild -project iina.xcodeproj ONLY_ACTIVE_ARCH=NO -scheme iina -configuration Nightly ${{ matrix.deployment_target }}
@@ -33,4 +35,3 @@ jobs:
         with:
           name: IINA
           path: ~/iina.tar.xz
-

--- a/other/download_libs.sh
+++ b/other/download_libs.sh
@@ -7,6 +7,7 @@ ARCH="universal"
 # github | iina (use iina to get the binary included in the latest release)
 YT_DLP_SOURCE="github"
 PARALLEL_DOWNLOADS=5
+SKIP_PLUGINS=false
 
 DYLIBS_DOWNLOAD_PATH="https://iina.io/dylibs/${ARCH}"
 YT_DLP_DOWNLOAD_PATH="https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp_macos"
@@ -19,7 +20,7 @@ BLUE='\033[0;34m'
 NC='\033[0m' # No Color
 
 # Reset in case getopts has been used previously in the shell.
-if ! OPTS=$(getopt -o "h": --long "arch:,yt-dlp-src:,parallel:,help": -n 'parse-options' -- "$@"); then
+if ! OPTS=$(getopt -o "h": --long "arch:,yt-dlp-src:,parallel:,skip-plugins,help": -n 'parse-options' -- "$@"); then
   echo -e "${RED}Failed parsing options.${NC}" >&2
   exit 1
 fi
@@ -31,6 +32,7 @@ printUsageHelp() {
   echo -e "    ${GREEN}$0 [--arch] <ARCH>:${NC}       Architecture to download dylibs for: universal | arm64 | x86_64"
   echo -e "    ${GREEN}$0 [--yt-dlp-src] <SRC>:${NC}  Source to download youtube-dl from: github | iina"
   echo -e "    ${GREEN}$0 [--parallel] <N>:${NC}      Number of parallel downloads (default: 5)"
+  echo -e "    ${GREEN}$0 [--skip-plugins]:${NC}      Skip downloading official plugins"
   echo
 }
 
@@ -79,6 +81,10 @@ while true; do
     fi
     PARALLEL_DOWNLOADS=$2
     shift 2
+    ;;
+  --skip-plugins)
+    SKIP_PLUGINS=true
+    shift
     ;;
   --)
     shift
@@ -162,6 +168,12 @@ curl -s -L "$YT_DLP_DOWNLOAD_PATH" -o "$YT_DLP_PATH" && echo -e "${GREEN}yt-dlp 
 chmod +x "$YT_DLP_PATH"
 
 mkdir -p "$PLUGIN_PATH"
+
+if [[ "$SKIP_PLUGINS" == true ]]; then
+  echo -e "${YELLOW}Skipping official plugin downloads.${NC}"
+  echo -e "${GREEN}All downloads completed.${NC}"
+  exit 0
+fi
 
 fetch_latest_plugin_asset() {
   local repo="$1"

